src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.cs

```diff
@@ -30,6 +30,29 @@ protected AbstractSyntaxClassificationService()
         public abstract string? GetSyntacticClassificationForIdentifier(SyntaxToken identifier);
 
         public async Task AddSemanticClassificationsAsync(
+            Document document,
+            ImmutableArray<TextSpan> textSpans,
+            ClassificationOptions options,
+            Func<SyntaxNode, ImmutableArray<ISyntaxClassifier>> getNodeClassifiers,
+            Func<SyntaxToken, ImmutableArray<ISyntaxClassifier>> getTokenClassifiers,
+            ArrayBuilder<PooledObject<SegmentedList<ClassifiedSpan>>> result,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                for (var i = 0; i < textSpans.Length; i++)
+                {
+                    AddSemanticClassifications(semanticModel, textSpans[i], getNodeClassifiers, getTokenClassifiers, result[i].Object, options, cancellationToken);
+                }
+            }
+            catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e, cancellationToken))
+            {
+                throw ExceptionUtilities.Unreachable();
+            }
+        }
+
+        public async Task AddSemanticToBeRemovedAsync(
```

Comment:
>AddSemanticToBeRemovedAsync [](http://example.com/codeflow?start=26&length=27)

without any context, this name is confusing. I'll probably figure it out, but it's definitely not intuitive what this is doing.